db/changes_view.go

```diff
@@ -66,26 +67,26 @@ func (dbc *DatabaseContext) getChangesInChannelFromView(
 			}
 			if options.ActiveOnly {
 				if !entry.IsRemoved() && entry.Flags&channels.Deleted == 0 {
-					// base.LogTo("Cache", "  Got view sequence #%d (%q / %q)", entry.Sequence, entry.DocID, entry.RevID)
-					entries = append(entries, entry)
+					activeEntryCount++
 				}
-			} else {
-				entries = append(entries, entry)
 			}
+			entries = append(entries, entry)
 			optMap["startkey"] = []interface{}{channelName, entry.Sequence + 1}
 		}
 
 		if options.ActiveOnly {
-			if len(entries) >= options.Limit || options.Limit == 0 {
+			if activeEntryCount >= options.Limit || options.Limit == 0 {
```

Comment:
I'm struggling to understand this code.  Can you add a comment to explain why it breaks out of the for loop in this case?

Also I think early breaks would be easier to understand .. eg,:

if !options.ActiveOnly {
   // No need to keep iterating, because...
   break
}

if options.ActiveOnly && activeEntryCount >= options.Limit || options.Limit == 0 {
   // no need to continue, because ..
   break
}

Reply:
Added comments to explain it.  Didn't use early breaks, because I think it makes the second if statement a lot harder to parse.